packages/app/src/renderer/features/current-game-overview.tsx

```diff
@@ -64,30 +67,41 @@ const getFactionMatrix = (gameData: GameData): string => {
 };
 
 const CurrentGameOverview: React.FC<Props> = ({ game }) => {
-  const [mapApiData, setMapApiData] = useState(null);
+  const dispatch = useDispatch();
+  const applicationCache = useSelector(selectCache);
+  const [requestState, setRequestState] = useState<"loading" | "idle" | "completed">("idle");
   const [mapWinLosses, setMapWinLosses] = useState<{ wins: number; losses: number }>();
   const [mapFound, setMapFound] = useState(false);
 
-  // listen map stats from firestore
   useEffect(() => {
-    //convertDateToMonthTimestamp(new Date())
-    const monthTimestamp =
-      Date.UTC(new Date().getUTCFullYear(), new Date().getUTCMonth() - 1, 1) / 1000;
-    //Date.UTC(new Date().getUTCFullYear(), new Date().getUTCMonth() - 1, 1) / 1000;
-    onSnapshot(doc(getFirestore(), `stats/month/${monthTimestamp}/`, "mapStats"), (doc) => {
-      setMapApiData(doc.data());
-    });
-  }, []);
-
-  useEffect(() => {
-    events.game_displayed();
-  }, [game]);
-
-  // update displayed map stats when gamedata changes
-  useEffect(() => {
-    // can only show valid data classic games
-    if (mapApiData) {
-      const result = findMapInApiData(mapApiData, game);
+    if (
+      !applicationCache.mapStats ||
+      !datesAreOnSameDay(new Date(applicationCache.mapStats.requestDate), new Date(Date.now()))
+    ) {
+      if (requestState === "idle") {
+        console.log("Request new mapstats data");
+        const monthTimestamp =
+          Date.UTC(new Date().getUTCFullYear(), new Date().getUTCMonth() - 1, 1) / 1000;
+        //Date.UTC(new Date().getUTCFullYear(), new Date().getUTCMonth() - 1, 1) / 1000;
+        setRequestState("loading");
+        getDoc(doc(getFirestore(), `stats/month/${monthTimestamp}/`, "mapStats")).then(
+          (docSnap) => {
+            if (docSnap.exists()) {
+              dispatch(
+                actions.setMapStatCache({
+                  requestDate: Date.now(),
+                  data: docSnap.data(),
+                }),
+              );
+              setRequestState("completed");
+            } else {
+              setRequestState("idle");
```

Comment:
Btw couldn't this trigger infinite loop which would be requesting the DB all the time in case the document wouldn't exist?  I would put it into completed state regardless and we should log an error.

Reply:
Hmm good point. the exists function is false when the data does not exist but what does that really mean? Should we attempt to request this again or give up.

I realized there is a mistake anyways ive implemented a then without a on rejected callback. This would likely trigger when there is no connection. And in this case I wanted to attempt to request again until the connection is back. That was the idea behind setting the request state back to idle when the file didnt exist.

Reply:
Ive opted to not try again if the request fails. It might result in not having the mapdata if you had a bad connection, but if you load into a new game it will have a chance to check again. This should be the save approach